target/linux/ath79/dts/qca9558_mikrotik_routerboard-92x.dtsi

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca955x.dtsi"
+
+/ {
+	aliases {
+		led-boot = &led_user;
+		led-failsafe = &led_user;
+		led-upgrade = &led_user;
+		serial0 = &uart;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user: user {
+			label = "mikrotik:green:user";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	ath10k-leds {
+		compatible = "gpio-leds";
+
+		wlan5g {
+			label = "mikrotik:green:wlan5g";
+			gpios = <&ath10k 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_nand_power {
+			gpio-export,name = "mikrotik:power:nand";
+			gpio-export,output = <0>;
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy4: ethernet-phy@4 {
+		reg = <4>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&phy4>;
+	pll-data = <0x8f000000 0xa0000101 0xa0001313>;
+
+	gmac-config {
+		device = <&gmac>;
+		rgmii-enabled = <1>;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "mikrotik,routerboot-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "routerboot";
+				reg = <0x0 0x0>;
+				read-only;
+			};
+
+			hard_config: hard_config {
+				read-only;
+			};
+
+			bios {
+				read-only;
```

Comment:
For the sake of consistency I would eventually add `size <0x1000>` here.

Reply:
Since this is only copied here, I'd do that separately.

Reply:
@f00b4r0 as said in the PR-message / commit-message the dtsi file is make by copying from the RB922.dts file without any change. If this parameter is missing a separate PR is more natural.

Reply:
The point is that not using the `size` keyword was OK-ish in a DTS file dedicated to a specific device where the size of the flash was known to "work" with this parameter missing as all partitions were essentially concatenated. Still, it really should be there and if this partition scheme is used (now via this DTSI) in a device where the flash chip is larger, things will break.

Please do fix this here.

@adrianschmutzler that's the reason (moving from DTS to DTSI) why I'm specifically requesting this be added now, I think this is acceptable.

Reply:
@f00b4r0 I will not change this a part of this patch, but if you just provide a separate patch I will throw it on top (via GitHub, patchwork, or directly to my mail address)

Reply:
@adrianschmutzler i can't do that while at work, but eventually I would only submit a one-liner exactly as suggested in my first comment. Please do feel free to add it in your tree, I'm not claiming any credit on such a trivial patch.

The current situation is in violation of the DTS specification for routerbootpart (as defined in the driver header). Thanks!

Reply:
Rephrasing: not in violation, rather it opens a door that shouldn't be opened in a DTSI, where this definition can result in an incorrectly sized bios partition. While this is not lethal and mostly cosmetic, I'd rather avoid having this copy-pasted by other contributors as-is. HTH

Reply:
@f00b4r0 Maybe just paste a commit title/message here, and I will take care of the content.

Reply:
ath79: routerboard-92x: ensure explicit bios size in DTSI

The missing "size" property was acceptable in the context of a single DTS as the underlying device is known to have a 64KB flash, and thus the bios partition fit exactly between the preceding and following ones.

However as this block has moved in a DTSI, for the sake of clarity and explicitness the size argument is added to ensure that if the flash happens to be larger than expected, the bios partition remains properly sized.

Reply:
Merged in 6e7f40dff3aa3fdbc012d0d9f79a3decdbe3735f.